src/Hedgehog/Gen.fs

```diff
@@ -69,20 +69,26 @@ module Gen =
     let map (f : 'a -> 'b) (g : Gen<'a>) : Gen<'b> =
         mapTree (Tree.map f) g
 
+#if !FABLE_COMPILER
     [<CompiledName("Map")>]
+#endif    
     let map2 (f : 'a -> 'b -> 'c) (gx : Gen<'a>) (gy : Gen<'b>) : Gen<'c> =
         bind gx <| fun x ->
         bind gy <| fun y ->
         constant (f x y)
 
+#if !FABLE_COMPILER
     [<CompiledName("Map")>]
+#endif    
     let map3 (f : 'a -> 'b -> 'c -> 'd) (gx : Gen<'a>) (gy : Gen<'b>) (gz : Gen<'c>) : Gen<'d> =
         bind gx <| fun x ->
         bind gy <| fun y ->
         bind gz <| fun z ->
         constant (f x y z)
 
+#if !FABLE_COMPILER
     [<CompiledName("Map")>]
+#endif    
```

Comment:
One last thing:

Say someone (who's not aware of this discussion) adds `map5` and adds also the attribute to it.

Is there an automated way (perhaps via a test we can run as part of the test-suite) to fail the build? (So that we don't forget to add also the compiler redirective.)

Reply:
So, the best think we can do is call fable compiler and is if it is happy with Hedgehog :-).

So i made a subfolder in tests called Fable.Tests which does only compile the test project which references Hedgehog.fsproj. I tried commenting out `#if !FABLE_COMPILER` and then it fails.

I also made two script to start the compiler `compile-with-fable.cmd` and `compile-with-fable.sh`. But, there is a new build dependency. Fable compiler runs on nodejs so the to call this scripts you have to install npm on the build maschine. I hope this is not a problem.

Reply:
Sorry. Something is still not working. Please, wait.

Reply:
Perhaps we can see how other projects deal with this. If they don't, perhaps we shouldn't bother for now(?) 😃

Reply:
I saw you added a few commits with tests. We can surely take those separately if you want, and keep this PR clean (only with those if compiler pragmas) and merge it.

Reply:
>Sorry. Something is still not working. Please, wait.

[No rush](http://butunclebob.com/ArticleS.UncleBob.SpeedKills) :+1: :wink:

Reply:
So, I am not excatly sure why my build in the morning did not work but I guess it is because I started docker container with linux shell to test the compile-with-fable.sh script and it changed my package-lock.json file. Now it works again, at lease on my windows maschine.

I also moved all the npm dependencies to development dependencies and removed a few npm libraries we do not need for just the compilation process.

I think this should do it.

Now it is up to you to tell me if you want the compile-with-fable.(cmd|sh) script to be called in the build process or not and how to do it.

If you want I can also move the tests to another branch and make another pull-request but I think it would be better to leave it like that so if someone sees the changes he can also see how to test them with fable.